https://github.com/NLnetLabs/ldns/pull/15

--- a/dnssec.c
+++ b/dnssec.c
@@ -23,6 +23,9 @@
 #include <openssl/rand.h>
 #include <openssl/err.h>
 #include <openssl/md5.h>
+#include <openssl/bn.h>
+#include <openssl/dsa.h>
+#include <openssl/rsa.h>
 #endif
 
 ldns_rr *
--- a/dnssec_sign.c
+++ b/dnssec_sign.c
@@ -17,6 +17,9 @@
 #include <openssl/rand.h>
 #include <openssl/err.h>
 #include <openssl/md5.h>
+#include <openssl/bn.h>
+#include <openssl/dsa.h>
+#include <openssl/rsa.h>
 #endif /* HAVE_SSL */
 
 ldns_rr *
--- a/host2str.c
+++ b/host2str.c
@@ -28,6 +28,12 @@
 #include <time.h>
 #include <sys/time.h>
 
+#ifdef HAVE_SSL
+#include <openssl/bn.h>
+#include <openssl/dsa.h>
+#include <openssl/rsa.h>
+#endif
+
 #ifndef INET_ADDRSTRLEN
 #define INET_ADDRSTRLEN 16
 #endif
--- a/keys.c
+++ b/keys.c
@@ -16,6 +16,9 @@
 #ifdef HAVE_SSL
 #include <openssl/ssl.h>
 #include <openssl/rand.h>
+#include <openssl/bn.h>
+#include <openssl/rsa.h>
+#include <openssl/dsa.h>
 #ifndef OPENSSL_NO_ENGINE
 # include <openssl/engine.h>
 #endif
