--- config.sub	2011-01-26 23:32:14.000000000 +0300
+++ config.sub	2011-09-08 10:52:18.493090000 +0400
@@ -1249,6 +1249,9 @@
 	pmac | pmac-mpw)
 		basic_machine=powerpc-apple
 		;;
+  nacl)
+    basic_machine=i686-pc
+    ;;
 	*-unknown)
 		# Make sure to match an already-canonicalized machine name.
 		;;
--- ssh.c	2011-09-23 14:56:43.092240000 +0400
+++ ssh.c	2011-09-23 14:56:59.310281000 +0400
@@ -235,7 +235,7 @@
  * Main program for the ssh client.
  */
 int
-main(int ac, char **av)
+ssh_main(int ac, char **av)
 {
 	int i, r, opt, exit_status, use_syslog;
 	char *p, *cp, *line, *argv0, buf[MAXPATHLEN], *host_arg;
--- channels.h	2012-06-07 10:40:48.000000000 +0400
+++ channels.h	2012-06-07 10:41:18.000000000 +0400
@@ -161,9 +161,9 @@

 /* default window/packet sizes for tcp/x11-fwd-channel */
 #define CHAN_SES_PACKET_DEFAULT	(32*1024)
-#define CHAN_SES_WINDOW_DEFAULT	(64*CHAN_SES_PACKET_DEFAULT)
+#define CHAN_SES_WINDOW_DEFAULT	(4*CHAN_SES_PACKET_DEFAULT)
 #define CHAN_TCP_PACKET_DEFAULT	(32*1024)
-#define CHAN_TCP_WINDOW_DEFAULT	(64*CHAN_TCP_PACKET_DEFAULT)
+#define CHAN_TCP_WINDOW_DEFAULT	(4*CHAN_TCP_PACKET_DEFAULT)
 #define CHAN_X11_PACKET_DEFAULT	(16*1024)
 #define CHAN_X11_WINDOW_DEFAULT	(4*CHAN_X11_PACKET_DEFAULT)

