update to newer mosh release

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -106,8 +106,8 @@ http_archive(
 
 new_git_repository(
     name = "mosh",
-    remote = "https://github.com/rpwoodbu/mosh.git",
-    commit = "cf73e1f8799b01ad1ed9731c6b3d239b68509222", # tag = "mosh-1.3.2"
+    remote = "https://github.com/mobile-shell/mosh",
+    commit = "bc73a26316ede2a79259d859f8ee309b32412420", # tag = "mosh-1.4.0"
     build_file = "BUILD.mosh",
 )
 
--- a/external/BUILD.mosh
+++ b/external/BUILD.mosh
@@ -93,7 +93,11 @@ cc_library(
 
 cc_library(
     name = "crypto_lib",
-    srcs = glob(["src/crypto/*.cc"]),
+    srcs = [
+        "src/crypto/ocb_internal.cc",
+        "src/crypto/base64.cc",
+        "src/crypto/crypto.cc",
+    ],
     hdrs = glob(["src/crypto/*.h"]),
     deps = [
         ":util_lib",
