Navigate back to connection dialog easier.

--- a/mosh_app/mosh_window.js
+++ b/mosh_app/mosh_window.js
@@ -289,6 +289,8 @@ mosh.CommandInstance.prototype.sendKeyboard_ = function(string) {
   if (this.running_) {
     const te = new TextEncoder();
     this.moshNaCl_.postMessage({'keyboard': Array.from(te.encode(string))});
+  } else if (string == 'c') {
+    document.location.replace('/plugin/mosh/mosh_client.html?profile-id=mosh');
   } else if (string == 'x') {
     window.close();
   } else if (string == 'r') {
@@ -346,6 +348,7 @@ mosh.CommandInstance.prototype.onCrash_ =
     mosh.CommandInstance.prototype.exit_ = function(output) {
   this.io.print('\r\n' + output + '\r\n');
   console.log(output);
+  this.io.print('Press "c" to go to connect dialog.\r\n');
   this.io.print('Press "r" to reconnect.\r\n');
   this.io.print('Press "x" to close the window.\r\n');
   this.running_ = false;
